examples/medplum-provider/src/components/encounter/EncounterHeader.tsx

```diff
@@ -0,0 +1,187 @@
+import { Box, Button, Flex, Group, Menu, Paper, SegmentedControl, Stack, Text, Modal } from '@mantine/core';
+import { formatDate, formatHumanName } from '@medplum/core';
+import { Encounter, HumanName, Practitioner } from '@medplum/fhirtypes';
+import { IconChevronDown } from '@tabler/icons-react';
+import { JSX, useState } from 'react';
+import { useDisclosure } from '@mantine/hooks';
+
+interface EncounterHeaderProps {
+  encounter: Encounter;
+  practitioner?: Practitioner | undefined;
+  onStatusChange?: (status: Encounter['status']) => void;
+  onTabChange?: (tab: string) => void;
+}
+
+export const EncounterHeader = (props: EncounterHeaderProps): JSX.Element => {
+  const { encounter, practitioner, onStatusChange, onTabChange } = props;
+  const [status, setStatus] = useState<Encounter['status']>(encounter.status);
+  const [activeTab, setActiveTab] = useState<string>('notes');
+  const [confirmOpened, { open: openConfirm, close: closeConfirm }] = useDisclosure(false);
+
+  const handleStatusChange = (newStatus: Encounter['status']): void => {
+    if (newStatus === 'cancelled') {
+      openConfirm();
+      return;
+    }
+
+    setStatus(newStatus);
+    onStatusChange?.(newStatus);
+  };
+
+  const confirmStatusChange = (): void => {
+    setStatus('cancelled');
+    onStatusChange?.('cancelled');
+    closeConfirm();
+  };
+
+  const handleTabChange = (tab: string): void => {
+    setActiveTab(tab);
+    onTabChange?.(tab);
+  };
+
+  const practitionerName = practitioner?.name?.[0]
+    ? formatHumanName(practitioner.name[0] as HumanName)
+    : 'Unknown Provider';
+  const formattedDate = formatDate(encounter.period?.start);
+  const encounterDetail = formattedDate ? `${formattedDate} · ${practitionerName}` : practitionerName;
+
+  const renderMenuItems = (): JSX.Element => {
+    if (status === 'planned') {
+      return (
+        <>
+          <Menu.Item onClick={() => handleStatusChange('arrived')}>Arrived</Menu.Item>
+          <Menu.Item onClick={() => handleStatusChange('in-progress')}>In Progress</Menu.Item>
+          <Menu.Item onClick={() => handleStatusChange('cancelled')}>Cancelled</Menu.Item>
+          <Menu.Divider />
+        </>
+      );
+    }
+
+    if (status === 'arrived') {
+      return (
+        <>
+          <Menu.Item onClick={() => handleStatusChange('in-progress')}>In Progress</Menu.Item>
+          <Menu.Item onClick={() => handleStatusChange('cancelled')}>Cancelled</Menu.Item>
+          <Menu.Divider />
+        </>
+      );
+    }
+
+    if (status === 'in-progress') {
+      return (
+        <>
+          <Menu.Item onClick={() => handleStatusChange('finished')}>Finished</Menu.Item>
+          <Menu.Item onClick={() => handleStatusChange('cancelled')}>Cancelled</Menu.Item>
+        </>
+      );
+    }
+
+    return <></>;
```

Comment:
Super minor, but this could probably be `undefined` or `null` rather than creating a new fragment.